### ISSUE TITLE
Huge number of retries of unsuccessful requests in athena

### ISSUE THREAD

**Issue by ebirukov** [awslabs/aws-glue-catalog-sync-agent-for-hive#11]
One request to the hive metastore to add a partition generates an infinite number of repeated requests to athena. Which seems to prevent all following requests from being run in athena
![изображение](https://user-images.githubusercontent.com/15173395/141052919-922a4d7c-f05c-4796-b7f2-705dcae2a777.png)

**Comment by ebirukov** [comment]
Any unhandled execution error causes execute statement in athena because the completed flag is false
https://github.com/awslabs/aws-glue-catalog-sync-agent-for-hive/blob/6dd72172d3a5969a0b559bad2503c76155e1771a/src/main/java/com/amazonaws/services/glue/catalog/HiveGlueCatalogSyncAgent.java#L165-L170

**Comment by IanMeyers** [comment]
Yes, this is expected as the error may be retryable. What is the base exception causing the create partition statement to fail?

**Comment by ebirukov** [comment]
> Yes, this is expected as the error may be retryable. What is the base exception causing the create partition statement to fail?

In my case,exception causing is table not found, but  in a different case, it could be any other and to be generate an infinite number of repeated requests to athena
